File: languages/python/bloock/entity/encryption/aes_decrypter.py
import bloock._bridge.proto.encryption_entities_pb2 as proto
from bloock.entity.encryption.decrypter import Decrypter
from bloock.entity.encryption.decrypter_args import DecrypterArgs


class AesDecrypter(Decrypter):
    def __init__(self, args: DecrypterArgs) -> None:
        super().__init__(alg=proto.A256GCM, args=args)

    def to_proto(self) -> proto.Decrypter:
        local_key = None
        if self.args.local_key is not None:
            local_key = self.args.local_key.to_proto()

        managed_key = None
        if self.args.managed_key is not None:
            managed_key = self.args.managed_key.to_proto()

        return proto.Decrypter(
            alg=self.alg,
            local_key=local_key,
            managed_key=managed_key,
        )

File: languages/php/Bloock/Entity/Encryption/AesDecrypter.php
<?php

namespace Bloock\Entity\Encryption;

class AesDecrypter implements Decrypter
{
    public int $alg;
    public DecrypterArgs $args;

    public function __construct(DecrypterArgs $args)
    {
        $this->alg = \Bloock\EncryptionAlg::A256GCM;
        $this->args = $args;
    }

    public function toProto(): \Bloock\Decrypter
    {
        $p = new \Bloock\Decrypter();
        $p->setAlg($this->alg);

        if ($this->args->localKey != null) {
            $p->setLocalKey($this->args->localKey->toProto());
        }

        if ($this->args->managedKey != null) {
            $p->setManagedKey($this->args->managedKey->toProto());
        }

        return $p;
    }
}